image/api_selector_test.go

```diff
@@ -204,13 +204,36 @@ func TestAPISelector_SelectDefaultWhenBadJSON(t *testing.T) {
 	assert.Error(t, err, "unexpected end of JSON input")
 }
 
+func TestAPISelector_SelectTrailingComma(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		fmt.Fprintf(w, testAPIServerString)
+	}))
+	defer ts.Close()
+
+	u, _ := url.Parse(ts.URL)
+
+	as := NewAPISelector(u)
+
+	actual, err := as.Select(&Params{
+		Infra:    "test,",
+		Language: "ruby,",
+		OsxImage: "meow,",
+		Dist:     "yosamitty,",
+		Group:    "dev,",
+		OS:       "osx,",
+	})
+	assert.Equal(t, actual, "default")
+	assert.Error(t, err, "tag \"a\" contained \",\", which is not supported by job-board -- check .travis.yml for trailing comma")
```

Comment:
You may be looking for [`assert.EqualError`](https://godoc.org/github.com/stretchr/testify/assert#EqualError).